instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py

```diff
@@ -195,12 +195,17 @@ def __call__(self, env, start_response):
 
         start_time = _time_ns()
 
-        token = context.attach(extract(env, getter=otel_wsgi.wsgi_getter))
+        token, ctx, span_kind = get_token_context_span_kind(
```

Comment:
I was thinking more like a wrapper function for `start_span` which internally takes care of checking server spans something like

```python
span = _start_internal_or_server_span(name, start_time, context_getter)
```

this function would accept whatever is necessary to figure out if an existing span is present or not and decide internally whether to create a server span or an internal span, and which context to use as parent.

Reply:
This function will need to return token also along with span.
```
start_internal_or_server_span(tracer, name, start_time, env, context_getter):
    ..
    create internal/server span
    ..
    
    return span, token
``` 
I am not sure if this is okay.
thinking of putting this function in `instrumentation/utils.py`


Also, we can shorten the recurring snippet if we want to skip the function
as the else part in the snippet is not necessary. Something like this:
```
token = context = None
span_kind = SpanKind.INTERNAL
if get_current_span() is INVALID_SPAN:
       context = extract(carrier, getter=carrier_getter)
       token = attach(context)
       span_kind = SpanKind.SERVER
```
Please share your thoughts.

Reply:
> This function will need to return token also along with span.

That sounds OK. It can return the span + token and let the caller store a reference to the token and detach it at a later time.

Reply:
> Also, we can shorten the recurring snippet if we want to skip the function

The main issue is that this is a very specific piece of logic strictly followed by all server instrumentation so I think it makes sense to encapsulate the logic in a re-usable function that is easily discoverable and changeable in a single place. Number of lines of code is not a big deal.

Reply:
Makes sense
Created wrapper for start_span.